src/scripts/main.js

```diff
@@ -1,5 +1,39 @@
 'use strict';
 
-document.addEventListener('click', e => {
-  // write code here
+document.addEventListener('click', event => {
+  const eventY = event.clientY;
+  const eventX = event.clientX;
+  const wall = document.querySelector('.wall');
+  const wallHeight = wall.clientHeight;
+  const wallWidth = wall.clientWidth;
+  const wallTopCoord = wall.getBoundingClientRect().top;
+  const wallLeftCoord = wall.getBoundingClientRect().left;
+  const wallY = wall.clientTop;
+  const wallX = wall.clientLeft;
+  const spider = document.querySelector('.spider');
+  const spiderHeight = spider.offsetHeight;
+  const spiderWidth = spider.offsetWidth;
+
+  function setSpiderCoords(eventCoord, wallSideCoord,
+    wallCoord, spiderSize, wallSize) {
+    let coord = eventCoord - wallSideCoord - wallCoord - spiderSize / 2;
+
+    if (coord < 0) {
+      coord = 0;
+    }
+
+    if (coord > wallSize - wallCoord - spiderSize) {
+      coord = wallSize - spiderSize;
+    }
+
+    return coord;
+  }
+
+  const spiderY = setSpiderCoords(eventY, wallTopCoord,
+    wallY, spiderHeight, wallHeight);
```

Comment:
```suggestion
  const spiderY = setSpiderCoords(
    eventY, 
    wallTopCoord,
    wallY, 
    spiderHeight, 
    wallHeight,
  );
```
each argument on its own line